src/gpg2john.c

```diff
@@ -382,6 +389,11 @@ int gpg2john(int argc, char **argv)
 			printf("%s:%s:::%s::%s\n", login, last_hash, gecos, filename);
 			MEM_FREE(last_hash);
 		}
+		if (!hash_generated) {
+			fprintf(stderr, "Error: No hash was generated for %s, ensure that the input file contains a single private key only.\n", filename);
+			// continue;
+			exit(EXIT_FAILURE);
+		}
```

Comment:
Not sure we want to exit, perhaps the message and `continue;` is better?

Reply:
I am still thinking about which option is better. That is why I have `// continue;` there.

This fprintf error message might get lost among the hashes being printed out in some cases. This is perhaps the only reason why I went with `exit` instead of `continue`.

Reply:
The hash output is usually redirected so shouldn't be a problem. Several other tools warn and continue, eg. office2john and rar2john. OTOH they are probably more often run with wildcards than this one.

Reply:
OK. I will use the `continue` approach.

I just noticed that `gpg2john` doesn't really handle multiple input files for some unknown reason(s). I am investigating currently.